### PR TITLE
Publish for tags and fix sdk build

### PR DESCRIPTION
**Pull request by schniedergers**

### CODE REVIEW

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.1.0-develop-2018111323-4-g028d0607
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=767615)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.1.0-develop-2018111323-4-g028d0607&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.1.0-develop-2018111323-4-g028d0607&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.1.0-develop-2018111323-4-g028d0607
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.1.0-develop-2018111323-4-g028d0607)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.1.0-develop-2018111323-4-g028d0607/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.1.0-develop-2018111323-4-g028d0607

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.1.0-develop-2018111323-5-gd26b1acd
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=767844)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.1.0-develop-2018111323-5-gd26b1acd&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.1.0-develop-2018111323-5-gd26b1acd&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.1.0-develop-2018111323-5-gd26b1acd
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.1.0-develop-2018111323-5-gd26b1acd)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.1.0-develop-2018111323-5-gd26b1acd/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.1.0-develop-2018111323-5-gd26b1acd

File: Jenkinsfile [hunk 1]
```diff
@@ -364,19 +377,31 @@ boolean shouldUploadToS3(Map args=[:]) {
...
+    if (args.context.isTagPush()) {
+        bucket = "swiftnav-artifacts-jenkins"
+        path += context.gitDescribe() + "/"
+        upload = true
     } else {
-        if (args.context.isBranchPush(branches: ['master', '*-release'])) {
-            bucket = "swiftnav-artifacts-jenkins"
-            path = "piksi_buildroot/" + args.context.gitDescribe() + "/"
+        if (args.context.isPrPush()) {
+            bucket = "swiftnav-artifacts-pull-requests-jenkins"
```
[the remark this answers is not in the document]
Comment (benjaminaltieri) [reply Jenkinsfile]: Did a quick check and I see the `UNPROTECTED`, protected (version tag only) and nano artifacts, but I don't see the `INTERNAL` build artifacts or `uImage.piksiv3_prod`

[the remark this answers is not in the document]
Comment (schniedergers) [reply Jenkinsfile]: Ah yes, I see the issue. Files are saved as artifacts, but the upload to S3 did not work as expected when specifying a dir:
```
Uploading buildroot/output/images/piksiv3_prod/ to s3://swiftnav-artifacts-pull-requests-jenkins/piksi_buildroot/pr-1006/v2.1.0-develop-2018111323-4-ga1c60ea09/v3/prod/ 
Upload complete
```

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.1.0-develop-2018111323-7-g2baf9e36
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=770069)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.1.0-develop-2018111323-7-g2baf9e36&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.1.0-develop-2018111323-7-g2baf9e36&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.1.0-develop-2018111323-7-g2baf9e36
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.1.0-develop-2018111323-7-g2baf9e36)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.1.0-develop-2018111323-7-g2baf9e36/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.1.0-develop-2018111323-7-g2baf9e36

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.1.0-develop-2018111323-8-g67698145
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=770193)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.1.0-develop-2018111323-8-g67698145&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.1.0-develop-2018111323-8-g67698145&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.1.0-develop-2018111323-8-g67698145
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.1.0-develop-2018111323-8-g67698145)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.1.0-develop-2018111323-8-g67698145/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.1.0-develop-2018111323-8-g67698145

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.1.0-develop-2018111820-9-g7c5d82ba

These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.

### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=776464)

### gnss-analysis results
At least one run must complete for these links to have data.

#### passfail"
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&summary_type=q50&groupby_key=firmware&firmware_versions=v2.1.0-develop-2018111820-9-g7c5d82ba&build_type=buildroot_pr&display_type=table)

#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/metrics_preset=detailed&scenario=live-roof-650-townsend-post&summary_type=q50&groupby_key=firmware&firmware_versions=v2.1.0-develop-2018111820-9-g7c5d82ba&build_type=buildroot_pr&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.1.0-develop-2018111820-9-g7c5d82ba
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.1.0-develop-2018111820-9-g7c5d82ba)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.1.0-develop-2018111820-9-g7c5d82ba/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.1.0-develop-2018111820-9-g7c5d82ba

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.1.0-develop-2018111820-11-gad4b642b

These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.

### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=777988)

### gnss-analysis results
At least one run must complete for these links to have data.

#### passfail"
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&summary_type=q50&groupby_key=firmware&firmware_versions=v2.1.0-develop-2018111820-11-gad4b642b&build_type=buildroot_pr&display_type=table)

#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/metrics_preset=detailed&scenario=live-roof-650-townsend-post&summary_type=q50&groupby_key=firmware&firmware_versions=v2.1.0-develop-2018111820-11-gad4b642b&build_type=buildroot_pr&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.1.0-develop-2018111820-11-gad4b642b
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.1.0-develop-2018111820-11-gad4b642b)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.1.0-develop-2018111820-11-gad4b642b/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.1.0-develop-2018111820-11-gad4b642b

File: Jenkinsfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (schniedergers) [reply Jenkinsfile]: Here's the list of artifacts on S3 for a PR after the fixes):
```
> aws s3 ls s3://swiftnav-artifacts-pull-requests-jenkins/piksi_buildroot/pr-1006/v2.1.0-develop-2018111820-9-gc91257973/v3/prod/
2018-11-21 15:33:48    4564179 PiksiMulti-DEV-v2.1.0-develop-2018111820-9.bin
2018-11-21 15:33:48     516463 PiksiMulti-FAILSAFE-v2.1.0-develop-2018111820-9.bin
2018-11-21 15:33:49   22115042 PiksiMulti-INTERNAL-v2.1.0-develop-2018111820-9.bin
2018-11-21 15:32:36   20092890 PiksiMulti-UNPROTECTED-v2.1.0-develop-2018111820-9.bin
2018-11-21 15:32:34   21582074 PiksiMulti-v2.1.0-develop-2018111820-9.bin
2018-11-21 15:33:51       8035 boot.bin
2018-11-21 15:58:47  537759032 piksi_sdk.txz
2018-11-21 15:33:50   21816431 uImage.piksiv3_prod

> aws s3 ls s3://swiftnav-artifacts-pull-requests-jenkins/piksi_buildroot/pr-1006/v2.1.0-develop-2018111820-9-gc91257973/nano/evt0/
2018-11-21 15:28:05   88080384 sdcard.img

```

**Pull request merged**